src/helm/proxy/tokenizers/tokenizer.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC, abstractmethod
+from dataclasses import asdict
+from typing import Callable, Any, Dict, List, Optional
+
+from helm.common.cache import Cache, CacheConfig
+from helm.common.request import wrap_request_time
+from helm.common.tokenization_request import (
+    TokenizationRequest,
+    TokenizationRequestResult,
+    DecodeRequest,
+    DecodeRequestResult,
+    TokenizationToken,
+)
+
+
+class Tokenizer(ABC):
```

Comment:
Also keep this a pure abstract interface, and move the logic into a subclass `CachingTokenizer`.